data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -628,130 +582,168 @@ impl TraceExporter {
-                let endpoint = Endpoint {
-                    url: self.get_agent_url(),
-                    ..self.endpoint.clone()
-                };
-                let mut headers: HashMap<&str, String> = header_tags.into();
-                headers.insert(DATADOG_SEND_REAL_HTTP_STATUS_STR, "1".to_string());
-                headers.insert(DATADOG_TRACE_COUNT_STR, chunks.to_string());
-                headers.insert(CONTENT_TYPE.as_str(), APPLICATION_MSGPACK_STR.to_string());
-
-                let strategy = RetryStrategy::default();
-                self.runtime.block_on(async {
-                    // Send traces to the agent
-                    let result =
-                        send_with_retry(&endpoint, payload, &headers, &strategy, None).await;
-
-                    // Send telemetry for the payload sending
-                    if let Some(telemetry) = &self.telemetry {
-                        if let Err(e) = telemetry.send(&SendPayloadTelemetry::from_retry_result(
-                            &result,
-                            payload_len as u64,
-                            chunks as u64,
-                        )) {
-                            error!("Error sending telemetry: {}", e.to_string());
+        let payload = match self.input_format {
+            TraceExporterInputFormat::V04 => match self.output_format {
+                TraceExporterOutputFormat::V04 => trace_utils::collect_trace_chunks(
+                    collection,
+                    &header_tags,
+                    &mut tracer_payload::DefaultTraceChunkProcessor,
+                    self.endpoint.api_key.is_some(),
+                    false,
+                )
+                .map_err(|e| {
+                    TraceExporterError::Deserialization(DecodeError::InvalidFormat(e.to_string()))
+                })?,
+                TraceExporterOutputFormat::V05 => trace_utils::collect_trace_chunks(
+                    collection,
+                    &header_tags,
+                    &mut tracer_payload::DefaultTraceChunkProcessor,
+                    self.endpoint.api_key.is_some(),
+                    true,
+                )
+                .map_err(|e| {
+                    TraceExporterError::Deserialization(DecodeError::InvalidFormat(e.to_string()))
+                })?,
+            },
+            TraceExporterInputFormat::V05 => match self.output_format {
+                TraceExporterOutputFormat::V05 => trace_utils::collect_trace_chunks(
+                    collection,
+                    &header_tags,
+                    &mut tracer_payload::DefaultTraceChunkProcessor,
+                    self.endpoint.api_key.is_some(),
+                    true,
+                )
+                .map_err(|e| {
+                    TraceExporterError::Deserialization(DecodeError::InvalidFormat(e.to_string()))
+                })?,
+                // TODO: APMSP-18190
+                _ => todo!(
```

Comment:
nitpick - since the next PR is going to remove the panics it's not a big deal but `unimplemented!()` or `unreachable!()` are probably more appropriate than `todo!()` for a lot of these. The end result is the same, a panic. But the intention is clearer.